packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -115,15 +126,47 @@ export const AllowTombstoneRequestHeaderKey = "allowTombstone"; // Belongs in th
 
 type PendingAliasResolve = (success: boolean) => void;
 
-interface FluidDataStoreMessage {
-	content: unknown;
-	type: string;
+/**
+ * Envelope for signals not intended for the container.
+ *
+ * @privateRemarks
+ * `clientBroadcastSignalSequenceNumber` might be added to the envelope by the container runtime.
+ * But it should not be provided to start with.
+ *
+ * @internal
+ */
+export type AddressedSignalEnvelope = IEnvelope<ISignalEnvelope["contents"]>;
+
+/**
+ * Kin of {@link @fluidframework/runtime-definitions#IFluidParentContext} with alternate
+ * submitMessage and submitSignal methods.
+ *
+ * @internal
+ * @privateRemarks Exposed per ChannelCollection testing and API extractor request
+ */
+export interface IFluidRootParentContext
```

Comment:
You'll need to rationalize splitting this out with the new `IFluidParentContextPrivate` added by #24465